### PR TITLE
Fix #12596

### LINKED ISSUE #12596: kube-proxy: userspace proxier should clean up iptables proxier's mess
If you switch from iptables back to userspace, all the rules are left behind.

@BenTheElder

### PR DESCRIPTION
**Pull request by BenTheElder**
Fix for https://github.com/kubernetes/kubernetes/issues/12596
Disconnect the pure-iptables proxy’s services chain when starting the userspace proxy.

### CODE REVIEW

File: pkg/proxy/userspace/proxier.go [hunk 1]
```diff
@@ -131,6 +131,8 @@ func createProxier(loadBalancer LoadBalancer, listenIP net.IP, iptables iptables
 	if proxyPorts == nil {
 		proxyPorts = newPortAllocator(util.PortRange{})
 	}
+	glog.V(2).Info("Tearing down pure-iptables proxy rules.")
```
Comment (BenTheElder) [inline pkg/proxy/userspace/proxier.go]: Self-Nit: I'm not sure if this should say "Errors are acceptable here." like we did in the iptables proxy.

**Comment by BenTheElder** [comment]
Rebased to try to make shippable happy.

File: pkg/proxy/userspace/proxier.go (hunk 1 above)
Comment (mikedanese) [inline pkg/proxy/userspace/proxier.go]: how often will this be called? I would suggest a higher log level

Comment (BenTheElder) [inline pkg/proxy/userspace/proxier.go]: Just once on startup. The log level here mimics the one in the iptables
Proxier, so if we change this we should change both.

Comment (thockin) [inline pkg/proxy/userspace/proxier.go]: I think the same message is appropriate

**Comment by thockin** [comment]
LGTM with nit on error message

File: pkg/proxy/userspace/proxier.go (hunk 1 above)
Comment (mikedanese) [inline pkg/proxy/userspace/proxier.go]: ok sounds good. I think it's okay to leave it

Comment (BenTheElder) [inline pkg/proxy/userspace/proxier.go]: I had to go out to move my sister in to her dorm. Will fix nit asap...
sometime later tonight.

**Comment by BenTheElder** [comment]
@thockin, updated.
squashing.

**Comment by thockin** [comment]
LGTM

@k8s-oncall priority for GSoC please

**Pull request merged**